pkg/distro/image_config.go

```diff
@@ -44,20 +44,22 @@ type ImageConfig struct {
 
 	// for RHSM configuration, we need to potentially distinguish the case
 	// when the user want the image to be subscribed on first boot and when not
-	RHSMConfig          map[subscription.RHSMStatus]*subscription.RHSMConfig
-	SystemdLogind       []*osbuild.SystemdLogindStageOptions
-	CloudInit           []*osbuild.CloudInitStageOptions
-	Modprobe            []*osbuild.ModprobeStageOptions
-	DracutConf          []*osbuild.DracutConfStageOptions
-	SystemdDropin       []*osbuild.SystemdUnitStageOptions
-	SystemdUnit         []*osbuild.SystemdUnitCreateStageOptions
-	Authselect          *osbuild.AuthselectStageOptions
-	SELinuxConfig       *osbuild.SELinuxConfigStageOptions
-	Tuned               *osbuild.TunedStageOptions
-	Tmpfilesd           []*osbuild.TmpfilesdStageOptions
-	PamLimitsConf       []*osbuild.PamLimitsConfStageOptions
-	Sysctld             []*osbuild.SysctldStageOptions
+	RHSMConfig    map[subscription.RHSMStatus]*subscription.RHSMConfig
+	SystemdLogind []*osbuild.SystemdLogindStageOptions
+	CloudInit     []*osbuild.CloudInitStageOptions
+	Modprobe      []*osbuild.ModprobeStageOptions
+	DracutConf    []*osbuild.DracutConfStageOptions
+	SystemdDropin []*osbuild.SystemdUnitStageOptions
+	SystemdUnit   []*osbuild.SystemdUnitCreateStageOptions
+	Authselect    *osbuild.AuthselectStageOptions
+	SELinuxConfig *osbuild.SELinuxConfigStageOptions
+	Tuned         *osbuild.TunedStageOptions
+	Tmpfilesd     []*osbuild.TmpfilesdStageOptions
+	PamLimitsConf []*osbuild.PamLimitsConfStageOptions
+	Sysctld       []*osbuild.SysctldStageOptions
+	// Do not use DNFConfig directly, call "DNFConfigOptions()"
 	DNFConfig           []*osbuild.DNFConfigStageOptions
+	DNFSetReleaseVerVar *bool
```

Comment:
Perhaps we could make these non-exported and export only `DNFConfigOptions()`. This, however, would require a custom Unmarshaler.

Reply:
Thanks for the suggestion, I'm inclined to do this in a followup.